crates/services/consensus_module/poa/src/signer.rs

```diff
@@ -0,0 +1,84 @@
+use anyhow::anyhow;
+use aws_sdk_kms::{
+    primitives::Blob,
+    types::{
+        MessageType,
+        SigningAlgorithmSpec,
+    },
+};
+use fuel_core_types::{
+    blockchain::{
+        block::Block,
+        consensus::{
+            poa::PoAConsensus,
+            Consensus,
+        },
+        primitives::SecretKeyWrapper,
+    },
+    fuel_vm::Signature,
+    secrecy::{
+        ExposeSecret,
+        Secret,
+    },
+};
+use std::ops::Deref;
+
+/// How the block is signed
+#[derive(Clone, Debug)]
+pub enum SignMode {
+    /// Blocks cannot be produced
+    Unavailable,
+    /// Sign using a secret key
+    Key(Secret<SecretKeyWrapper>),
+    /// Sign using AWS KMS
+    Kms {
+        key_id: String,
+        client: aws_sdk_kms::Client,
+    },
+}
+
+impl SignMode {
+    /// Is block sigining (production) available
+    pub fn is_available(&self) -> bool {
+        !matches!(self, SignMode::Unavailable)
+    }
+
+    /// Sign a block
+    pub async fn seal_block(&self, block: &Block) -> anyhow::Result<Consensus> {
+        let block_hash = block.id();
+        let message = block_hash.into_message();
+
+        // https://docs.aws.amazon.com/kms/latest/developerguide/asymmetric-key-specs.html
+        // const AWS_KEY_TYPE: KeySpec = KeySpec::ECC_SECG_P256K1;
+
+        let poa_signature = match self {
+            SignMode::Unavailable => return Err(anyhow!("no PoA signing key configured")),
+            SignMode::Key(key) => {
+                // The length of the secret is checked
+                let signing_key = key.expose_secret().deref();
+                Signature::sign(signing_key, &message)
+            }
+            SignMode::Kms { key_id, client } => {
+                let reply = client
+                    .sign()
+                    .key_id(key_id)
+                    .signing_algorithm(SigningAlgorithmSpec::EcdsaSha256)
+                    .message_type(MessageType::Digest)
+                    .message(Blob::new(*message))
+                    .send()
+                    .await?;
+                let signature_der = reply
+                    .signature
+                    .ok_or_else(|| anyhow!("no signature returned from AWS KMS"))?
+                    .into_inner();
```

Comment:
Will this also raise up errors from the SDK? If not, we should surface those for logging - it will make debugging permission or config issues _much_ easier.

Reply:
We do capture the SDK errors. In this specific case though, there's no error avaible from the SDK. I'm not even sure if this branch can occur; the AWS SDK bindings are poorly documented. It is quite likely that this field can never be `None`, as any error would be captured by the `.send()?` above.